Verbale Esterno 2024-07-09 (#147)
Argo-swe/Docs: sources/documents/2_RTB/Verbali/Esterni/VerbaleEsterno_2024-07-09/capitolo-riunione.tex

\par Dopo l'arrivo della Proponente il gruppo espone il motivo dell'incontro per presentare la versione del Proof of Concept, realizzata con \glossario{VueJS} come strumento per il \glossario{front-end} e \glossario{FastAPI} per il \glossario{back-end}.
Il gruppo spiega quelli che sono stati i cambiamenti fatti dall'ultimo incontro, soprattutto riguardanti delle correzioni nella gestione delle chiavi esterne che sono ora gestite come chiavi primarie.
Vengono poi spiegati i modelli in lingua italiana utilizzati per effettuare dei test per la generazione del \glossario{prompt}. In particolare sono stati testati i modelli m-bertino e m-marco di Efederici. I test sono stati effettuati con \glossario{LMstudio} utilizzando una macchina potente e modelli in locale utilizzando \glossario{Llama3}. 
I test sono stati eseguiti con lo stesso \glossario{prompt}, ripetuti per un numero di volte e i risultati salvati in un documento di testo per poi eseguire un confronto.

\par Viene poi avviata la condivisione con la quale la Proponente ha potuto visionare la versione del Proof of Concept. Viene presentata la gestione dell'autenticazione con la quale l'utente può ricevere i privilegi da admin. 

> Da qui viene incollato una richiesta sufficientemente complessa che genera un \glossario{prompt} e viene mostrato il risultato ottenuto.

Reviewer: ```suggestion
Successivamente viene incollato una richiesta preselezionata dal team e sufficientemente complessa che genera un \glossario{prompt} e viene mostrato il risultato ottenuto.
```

Il risultato viene poi copiato per essere incollato su \glossario{ChatGPT} e \glossario{LMsys} per ottenere un confornto sui risultati ottenuti dalla generazione della query su vari modelli.

> \par La Proponente trova interessante il risultato ottenuto da \glossario{ChatGPT}, in particolare l'utilizzo di un NOT EXIST e che la query risultato fosse abbastanza ridotta. Il gruppo commenta osservando che, dai test svolti, \glossario{ChatGPT} sembri essere quello che ritorna le query più ridotte.

Reviewer: ```suggestion
\par La \glossario{Proponente} trova interessante il risultato ottenuto da \glossario{ChatGPT}, in particolare l'utilizzo di un NOT EXIST e che la query risultato fosse abbastanza ridotta. Il gruppo commenta osservando che, dai test svolti, \glossario{ChatGPT} sembri essere quello che ritorna le  \glossario{query} più ridotte.
```

> La Proponente chiede di testare il \glossario{prompt} su altri modelli, utilizzando \glossario{LMsys}. Vengono svolti test diversi con modelli che propongono risultati simili, ma con delle divergenze nell'espressione della query. Ad ogni test, la Proponente esamina con il gruppo il significato e la correttezza delle query generate, evidenziando come, di modello in modello, ci siano delle sezioni dove il modello sembri confondersi data la complessità della ricerca.

Reviewer: ```suggestion
La \glossario{Proponente} chiede di testare il \glossario{prompt} su altri modelli, utilizzando \glossario{LMsys}. Vengono svolti test diversi con modelli che propongono risultati simili, ma con delle divergenze nell'espressione della \glossario{query}. Ad ogni test, la \glossario{Proponente} esamina con il gruppo il significato e la correttezza delle  \glossario{query} generate, evidenziando come, di modello in modello, ci siano delle sezioni dove il modello sembra confondersi data la complessità della ricerca.
```

> La Proponenete chiede al gruppo se sono stati svolti dei test anche su un \glossario{database} locale: il gruppo afferma di aver svolto dei test sempre con una stessa richiesta e che questa ha portato i risultati richiesti.

Reviewer: ```suggestion
La \glossario{Proponente} chiede al gruppo se sono stati svolti dei test anche su un \glossario{database} locale: il gruppo afferma di aver svolto dei test sempre con una stessa richiesta e che questa ha portato i risultati richiesti.
```

> La Proponente evidenzia come non ci siano errori palesi, tuttavia nelle query generate si verificano situazioni che potrebbero portare a delle condizioni di errore e chiede al gruppo di svolgere più test provando con dei \glossario{database} in locale.

Reviewer: ```suggestion
La \glossario{Proponente} evidenzia come non ci siano errori strutturali evidenti nella generazione delle \glossario{query}, tuttavia da questo esempio si evince che potrebbero verificarsi condizioni di errore e viene chiesto al gruppo di svolgere più test provando con dei \glossario{database} in locale.
```

> \par Il gruppo ritorna al Proof of Concept per illustrare le funzionalità rimanenti. Viene mostrato il \glossario{debug} che illustra in un \glossario{log} i valori rilevanti e da la possibilità di scaricarlo in un formato testuale. Inoltre viene mostrata la sezione riguardante la visualizzazione, l'aggiunta, la modifica e l'eliminazione dei \glossario{dizionari dati}.

Reviewer: ```suggestion
\par Al termine dei test con gli LLM il gruppo ha riportato l'attenzione al Proof of Concept per illustrare le funzionalità rimanenti. Viene mostrato il \glossario{debug} che illustra in un \glossario{log} i valori rilevanti e da la possibilità di scaricarlo in un formato testuale. Inoltre viene mostrata la sezione riguardante la visualizzazione, l'aggiunta, la modifica e l'eliminazione dei \glossario{dizionari dati}.
```

> Viene poi effettuato il logout per mostrare come l'utente interagisce con l'applicativo: viene mostrata la pagina della chat con la quale l'utente può interagire inviando richieste in linguaggio naturale per ottenere il \glossario{prompt}. Successivamente vengono mostrate le impostazioni della chat, con le quali l'utente può selezionare il \glossario{dizionario dati}, vederne un'anteprima con una descrizione delle tabelle che contiene, selezionare il \glossario{DBSM} e la lingua di inserimento.

Reviewer 2: ```suggestion
Viene poi effettuato il logout per mostrare come l'utente interagisce con l'applicativo: viene mostrata la pagina della chat con la quale l'utente può interagire inviando richieste in linguaggio naturale per ottenere il \glossario{prompt}. Successivamente vengono mostrate le impostazioni della chat, con le quali l'utente può selezionare il \glossario{dizionario dati}, vederne un'anteprima con una descrizione delle tabelle che contiene, selezionare il \glossario{DBMS} e la lingua di inserimento.
```

Infine vengono illustrate le impostazioni di accessibilità con le quali è possibile cambiare la lingua dell'applicazione, la dimensione del font e il tema della pagina.

> \par La Proponente chiede se il gruppo è pronto per affrontare la prima revisione e suggerisce che ogni membro si prepari qualcosa da dire o esporre, durante la revisione. Inoltre consiglia di effettuare più test su \glossario{database}, sfruttandone uno in locale.

Reviewer 2: ```suggestion
\par La Proponente chiede se il gruppo è pronto per affrontare la prima revisione e suggerisce che ogni membro si prepari qualcosa da dire o esporre, durante la revisione. Inoltre rimarca l'utilità di effettuare più test su \glossario{database}, sfruttandone uno in locale.
```

Reviewer 2: Rifrasato in quanto riferimento alla discussione avuta in precedenza

> \par La Proponente chiede se il gruppo è pronto per affrontare la prima revisione e suggerisce che ogni membro si prepari qualcosa da dire o esporre, durante la revisione. Inoltre consiglia di effettuare più test su \glossario{database}, sfruttandone uno in locale.

Reviewer: ```suggestion
\par La \glossario{Proponente} chiede se il gruppo è pronto per affrontare la prima revisione e suggerisce che ogni membro si prepari qualcosa da dire o esporre, durante la revisione. Inoltre consiglia di effettuare più test su \glossario{database}, sfruttandone uno in locale.
```

\subsubsection{Conclusioni}

> \par Il gruppo conclude con alcune domande per la Proponente in vista del Proof of Concept

Reviewer: ```suggestion
\par Il gruppo conclude con alcune domande per la \glossario{Proponente} in vista del Proof of Concept
```

\par \textbf{Domanda:} Ci siamo accorti come alcuni modelli tendano a mettere delle parentesi in più o ad effettuare altre operazioni rispetto ad altri modelli. È normale? \\

> \textbf{Risposta:} Sì, è normale. I \glossario{database} hanno modi di scrivere diversi, come ad esempio il segno di "diverso da" che in alcuni viene presentato in stile "SQL" e in altri in stile "C++": la disuguaglianza quindi ad esempio, può essere motivo di confusione per il modello che procede a cercare quali \glossario{database} usano quella notazione e scrive altre informazioni di conseguenza.

Reviewer 2: ```suggestion
\textbf{Risposta:} Sì, è normale. I \glossario{database} hanno modi di scrivere diversi, come ad esempio il segno di "diverso da" che in alcuni viene presentato in stile "SQL" e in altri in stile "C++": la disuguaglianza quindi ad esempio, può essere motivo di confusione per il modello che procede a cercare quali \glossario{database} usano quella notazione e scrive altre informazioni di conseguenza. Anche l'utilizzo di "LIKE" rispetto a controlli di uguaglianza stretta sono adeguati, in quanto meno stringenti nel confronto tra stringhe.
```

> \par \textbf{Domanda:} Il Professor Cardin ci ha detto che è necessario avere una notazione quando un \glossario{database} è indicizzato. Lei cosa ne pensa a riguardo? Pensa sia necessario? \\

Reviewer 2: ```suggestion
\par \textbf{Domanda:} Nel considerare la performance di una query abbiamo considerato se agiungere una notazioneper descrivere gli indici di un \glossario{database}. Lei cosa ne pensa a riguardo? Ritiene sia necessario? \\
```

> \textbf{Risposta:} Non è necessario, poiché avendo quest'informazione l'unica cosa che cambia è la struttura e non la correttezza delle query, che è quello che chiediamo al fine del progetto.

Reviewer 2: ```suggestion
\textbf{Risposta:} Non è necessario, poiché avendo quest'informazione l'unica cosa che cambia è la struttura e non la correttezza delle query, che è quello che chiediamo al fine del progetto. Va considerato inoltre il rischio che aggiungere l'informazione nel prompt lo "inquini", inducendo il Large Language Model a query meno corrette.
```

> \par Il gruppo conclude la riunione ringraziando la Proponente per la disponibilità. La Proponente resta in attesa di notizie sul risultato della \RTB.

Reviewer: ```suggestion
\par Il gruppo conclude la riunione ringraziando la \glossario{Proponente} per la disponibilità. La Proponente resta in attesa di notizie sul risultato della \RTB.
```
